### ISSUE TITLE
Pivot Table collapse/expand rows doesn't work for single question in Public/Embedding

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#21847]
**Describe the bug**
It is not possible to collapse/expand rows of Pivot Table shared via Public or Embedding as single question. It works via dashboards.

**To Reproduce**
1. Question > Sample > Orders - summarize Count grouped by CreatedAt:Year, People.Source and Products.Category
![image](https://user-images.githubusercontent.com/1447303/164429461-89fcc62c-747f-40fc-ac7d-aa0e6bc9b607.png)
2. Change visualization to Pivot Table - save
3. Enabled Public Sharing (or Embedding)
4. Go to the shared address and try to collapse/expand rows - nothing happens and no errors in the browser console
![image](https://user-images.githubusercontent.com/1447303/164429038-3e4420a0-dbf3-4352-a2c5-2bc6e49c5800.png)

**Expected behavior**
Same behavior as adding the question on a dashboard and sharing the dashboard, which allows collapsing/expanding rows.

**Information about your Metabase Installation:**
Tested 0.39.0 thru 0.43.0-rc1

**Comment by urkle** [comment]
I noticed this as well.  It is quite odd. what seems to be happening though is that the "collapsed" node state is being saved to the query itself.  As I had collapsed them all while making the query and saved it like that, and when loading it via the public URL, all the nodes were collapsed and I was unable to access the detail data in the public URL.

The public view IMHO should be able to collapse data for easier viewing. And the prevention of accessing collapsed nodes should be an opt-in functionality if that was by design, as it is very unexpected.
